aten/src/ATen/native/cuda/TensorModeKernel.cu

```diff
@@ -140,7 +140,9 @@ void calculate_mode(
   // location of the buffer at the innermost dimension that we are going
   // to calculate the mode for --> we do this by manually doing the stride
   // calculations to get an offset
-  scalar_t* data = self.data_ptr<scalar_t>();
+  //
+  // We sort the data below, so we need this to be mutable.
+  scalar_t* data = self.mutable_data_ptr<scalar_t>();
```

Comment:
The modification here is correct, but the extra comment you added is not so clear. Naively, you would expect mode not to modify its input self. It turns out self is guaranteed to be a clone as enforced by

```
    // If transposed is already contiguous, it will return a tensor with the
    // same storage. So, since we do not want to modify self, we clone it.
    if (transposed.is_same(contiguous)) {
      contiguous = contiguous.clone();
    } 
```

This would have been immediately convincing to me

Reply:
Ah, I understand. I added a note and referenced it here. Is this better?

Reply:
yep, looks good!